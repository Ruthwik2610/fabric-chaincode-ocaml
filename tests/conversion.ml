let () =
  let timestamp =
    FabricChaincodeShim.Timestamp.Google.Protobuf.Timestamp.make ~seconds:42 ()
  in
  Printf.printf "%i\n" timestamp.seconds;
  let encoded =
    FabricChaincodeShim.Timestamp.Google.Protobuf.Timestamp.to_proto timestamp
    |> Ocaml_protoc_plugin.Writer.contents
  in
  Printf.printf "%S\n" encoded;
  let decoded =
    Ocaml_protoc_plugin.Reader.create encoded
    |> FabricChaincodeShim.Timestamp.Google.Protobuf.Timestamp.from_proto
    |> Result.get_ok
  in
  Printf.printf "%i\n" decoded.seconds
