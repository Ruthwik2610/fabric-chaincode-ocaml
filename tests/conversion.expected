42
"\b*"
42
